bluemira/equilibria/fem_fixed_boundary/equilibrium.py

```diff
@@ -21,238 +21,325 @@
+    msh_to_xdmf(mesh_name_msh, dimensions=(0, 2), directory=directory)
+    return import_mesh(mesh_filename, directory=directory, subdomains=True)[0]
+
+
+def _update_delta_kappa(
+    params: PlasmaFixedBoundaryParams,
+    kappa_95: float,
+    kappa95_t: float,
+    delta_95: float,
+    delta95_t: float,
+    relaxation: float,
+) -> float:
+    """
+    Recalculate Delta and Kappa and calculate the iteration error
+    """
+    err_delta = abs(delta_95 - delta95_t) / delta95_t
+    err_kappa = abs(kappa_95 - kappa95_t) / kappa95_t
+    iter_err = max(err_delta, err_kappa)
+
+    # calculate the new kappa_u and delta_u
+    kappa_u_0 = params.kappa_u
+    delta_u_0 = params.delta_u
+
+    kappa_u = (1 - relaxation) * kappa_u_0 * (
+        kappa95_t / kappa_95
+    ) + relaxation * kappa_u_0
+    delta_u = (1 - relaxation) * delta_u_0 * (
+        delta95_t / delta_95
+    ) + relaxation * delta_u_0
+
+    bluemira_debug(
+        "Previous shape parameters:\n\t"
+        f"{kappa_u_0=:.3e}, {delta_u_0=:.3e}\n"
+        "Recalculated shape parameters:\n\t"
+        f"{kappa_u=:.3e}, {delta_u=:.3e}\n\n"
+        f"|Target - Actual|/Target = {err_delta:.3e}\n"
+        f"|Target - bluemira|/Target = {err_kappa:.3e}\n"
+    )
+    params.kappa_u = kappa_u
+    params.delta_u = delta_u
+
+    return iter_err
+
+
+def solve_transport_fixed_boundary(
+    plasma_parameterisation: Type[GeometryParameterisation],
+    params: PlasmaFixedBoundaryParams,
+    transport_solver: CodesSolver,
+    transport_params: ParameterFrame,
+    gs_solver: FemGradShafranovFixedBoundary,
+    kappa95_t: float,
+    delta95_t: float,
+    lcar_mesh: float = 0.15,
+    max_iter: int = 30,
+    iter_err_max: float = 1e-5,
+    relaxation: float = 0.2,
+    transport_run_mode: Union[str, RunMode] = "run",
+    mesh_filename: str = "FixedBoundaryEquilibriumMesh",
+    plot: bool = False,
 ):
     """
     Solve the plasma fixed boundary problem using delta95 and kappa95 as target
-    values and iterating on PLASMOD to have consistency with pprime and ffprime.
+    values and iterating on a transport solver to have consistency with pprime
+    and ffprime.
 
     Parameters
     ----------
-    builder_plasma: Builder
-        Plasma poloidal cross section builder object
-    params: Configuration
-        Parameters to use in the PLASMOD solve
-    build_config: dict
-        Build configuration to use in the PLASMOD solve
-    gs_options: dict
-        Set of options used to set up and run the FemGradShafranovFixedBoundary
-    delta95_t: float
-        Target value for delta at 95%
+    plasma_parameterisation: Type[GeometryParameterisation]
+        Geometry parameterisation class for the plasma
+    params: PlasmaFixedBoundaryParams
+        dataclass of variables to use in the PLASMOD solve
+    transport_solver: CodesSolver
+        Transport Solver to call
+    transport_params: ParameterFrame
+        ParameterFrame for transport solver
```

Comment:
I tried that. Because of plasmod overwriting things it ends up oscillating between two solutions whereas if I have a separate object I always have the parameters to set the solver to, explicitly only updating a couple of parameters

Reply:
Could we do a copy instead?

Reply:
Hmm, yeah I suppose, I will give it a go

Reply:
Done something let me know :)